src/System.Xml.XmlSerializer/src/System/Xml/Serialization/Compilation.cs

```diff
@@ -305,21 +303,27 @@ public override int GetHashCode()
 
     internal class TempAssemblyCache
     {
-        private Hashtable _cache = new Hashtable();
+        private Dictionary<TempAssemblyCacheKey, TempAssembly> _cache = new Dictionary<TempAssemblyCacheKey, TempAssembly>();
```

Comment:
Ok.  It sounds like the change is good then... but there's still a threading problem to be dealt with subsequently.